cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -460,13 +460,19 @@ public void init(String channelId, @Nullable final NotificationReceivedListenerB
 
     // Check if CleverPush SharedPreferences is created and contains the key channelId
     SharedPreferences sharedPreferences = SharedPreferencesManager.getSharedPreferences(context);
-    boolean containsChannelId = sharedPreferences.contains("channelId");
+    boolean containsChannelId = sharedPreferences.contains(CleverPushPreferences.CHANNEL_ID);
+    boolean containsAppInstallationDate = sharedPreferences.contains(CleverPushPreferences.APP_INSTALLATION_DATE);
 
     // If CleverPush SharedPreferences is not created or does not contain the key channelId, call migrateSharedPreferences
     if (!containsChannelId) {
       SharedPreferencesManager.migrateSharedPreferences(context);
     }
 
+    if (!containsAppInstallationDate) {
+      String installationDate = new SimpleDateFormat(DATE_FORMAT_ISO, Locale.getDefault()).format(new Date());
```

Comment:
Please fallback to `CleverPushPreferences.SUBSCRIPTION_CREATED_AT` if it already defined.
Otherwise all existing Subscriptions will set (current date) as the installation date.

Reply:
@gbhrdt We are not setting value in `CleverPushPreferences.SUBSCRIPTION_CREATED_AT` anywhere in SDK.